plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3306,23 +3309,31 @@ public void testOptimize()
         assertUpdate("DROP TABLE " + tableName);
     }
 
-    private List<String> getActiveFiles(String tableName)
+    private Set<String> getActiveFiles(String tableName)
     {
         return computeActual(format("SELECT file_path FROM \"%s$files\"", tableName)).getOnlyColumn()
                 .map(String.class::cast)
-                .collect(toImmutableList());
+                .collect(toImmutableSet());
     }
 
-    private List<String> getAllDataFilesFromTableDirectory(String tableName)
+    private Set<String> getAllDataFilesFromTableDirectory(String tableName)
             throws IOException
     {
         String schema = getSession().getSchema().orElseThrow();
         Path tableDataDir = getDistributedQueryRunner().getCoordinator().getBaseDataDir().resolve("iceberg_data").resolve(schema).resolve(tableName).resolve("data");
         try (Stream<Path> list = Files.list(tableDataDir)) {
             return list
+                    .flatMap(path -> {
+                        try {
+                            return Files.walk(path).filter(Files::isRegularFile);
+                        }
+                        catch (IOException e) {
```

Comment:
This should not occur. If it does, something is wrong and we should fail

Reply:
yes I know but as IOException is a checked one and it was inside the flatMap I had to do something with it..